header/p2p/peer_tracker.go

```diff
@@ -12,23 +13,48 @@ import (
 
 type peerTracker struct {
 	sync.RWMutex
+
+	ctx    context.Context
+	cancel context.CancelFunc
+
 	connectedPeers map[peer.ID]*peerStat
 	// we cache the peer once they disconnect,
 	// so we can guarantee that peerQueue will only contain active peers
 	disconnectedPeers map[peer.ID]*peerStat
 
 	host host.Host
+
+	gcPeriod           time.Duration
+	maxAwaitingTime    time.Duration
+	defaultScore       float32
+	maxPeerTrackerSize int
+
+	done chan struct{}
 }
 
-func newPeerTracker(h host.Host) *peerTracker {
+func newPeerTracker(
+	h host.Host,
+	gcPeriod, maxAwaitingTime time.Duration,
+	defaultScore float32,
+	maxPeerTrackerSize int,
+) *peerTracker {
+	ctx, cancel := context.WithCancel(context.Background())
 	return &peerTracker{
-		disconnectedPeers: make(map[peer.ID]*peerStat),
-		connectedPeers:    make(map[peer.ID]*peerStat),
-		host:              h,
+		ctx:                ctx,
+		cancel:             cancel,
+		disconnectedPeers:  make(map[peer.ID]*peerStat),
+		connectedPeers:     make(map[peer.ID]*peerStat),
+		host:               h,
+		gcPeriod:           gcPeriod,
+		maxAwaitingTime:    maxAwaitingTime,
+		defaultScore:       defaultScore,
+		maxPeerTrackerSize: maxPeerTrackerSize,
+		done:               make(chan struct{}, 2),
 	}
 }
 
-func (p *peerTracker) track(ctx context.Context) {
+func (p *peerTracker) track() {
+	go p.gc()
```

Comment:
can you please spawn garbage collection from `ex.Start` ? makes it more clear what routines are spawned from start of exchange. I don't like spawning long-lived routines from other long-lived routines.